WordPressKit/WordPressAPIError.swift

```diff
@@ -0,0 +1,43 @@
+import Foundation
+
+public enum WordPressAPIError<EndpointError>: Error where EndpointError: LocalizedError {
+    static var unknownErrorMessage: String {
+        NSLocalizedString(
+            "wordpress-api.error.unknown",
+            value: "Something went wrong, please try again later.",
+            comment: "Error message that describes an unknown error had occured"
+        )
+    }
+
+    /// Can't encode the request arguments into a valid HTTP request. This is a programming error.
+    case requestEncodingFailure
+    /// Error occured in the HTTP connection.
+    case connection(URLError)
+    /// The API call returned an error result. For example, an OAuth endpoint may returns an 'incorrect username or password' error, an upload media endpoint may return an 'unsupported media type' error.
+    case endpointError(EndpointError)
+    /// WordPress.com returned an HTTP response that WordPressKit can't parse.
```

Comment:
What about things that aren't WordPress.com? :)

Reply:
I missed this particular comment, which is incorrect now. Because the new `WordPressAPIError` is for all API errors, not just WP.com ones. Updated in https://github.com/wordpress-mobile/WordPressKit-iOS/pull/653/commits/383490efb89a24820b704b7232cab97796f52d17